pkg/ebpf/telemetry/errors_collector_linux.go

```diff
@@ -8,40 +8,46 @@
 package telemetry
 
 import (
-	"fmt"
 	"syscall"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/sys/unix"
-
-	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 const (
 	maxErrno    = 64
 	maxErrnoStr = "other"
-
-	ebpfMapTelemetryNS    = "ebpf_maps"
-	ebpfHelperTelemetryNS = "ebpf_helpers"
 )
 
+// A singleton instance of the ebpf telemetry struct. Used by the collector and the ebpf managers (via ErrorsTelemetryModifier).
+var errorsTelemetry ebpfErrorsTelemetry
+
 // EBPFErrorsCollector implements the prometheus Collector interface
 // for collecting statistics about errors of ebpf helpers and ebpf maps operations.
 type EBPFErrorsCollector struct {
-	T                *EBPFTelemetry
+	T                ebpfErrorsTelemetry
```

Comment:
Is there a reason to have `EBPFErrorsCollector` and `EBPFTelemetry` as separate types? They are quite coupled together.

Reply:
that is the original design and implementation, I don't want to make those changes in this PR, but me and Usama can take a look how to simplify this package, it is indeed very complex